PORTING.md

```diff
@@ -0,0 +1,354 @@
+
+If you want to add a development board for an existing device, have a look at [`xpcc/architecture/platform/board`](https://github.com/roboterclubaachen/xpcc/tree/develop/src/xpcc/architecture/platform/board).
+
+## Porting a new device from AVR or STM32
+
+It may be as easy as adding a new device file, but if it's a brand new device, the device headers also need to be updated. When porting a new series (for example the STM32L4), a lot of code can be reused, but there may be very important differences between the two that are not immediately obvious to someone who isn't that familiar with STM32 devices.
+
+We recommend writing us an email, or opening up an issue on GitHub.
+
+## Porting a new device from a new vendor
+
+Note: We assume this new device is Cortex-M based. For other CPU architectures, send us an email.
+
+Porting to a new vendor requires you to:
+
+1. Add a bunch of header files for the new device.
+2. Add a device files for the new device.
+3. Make the build system aware of the new device classes.
+4. Adapt the Cortex-M startup- and linkerscripts.
+5. Compile a minimal example.
+6. Write a basic GPIO and delay driver.
+7. Compile an LED blinking example.
+
+### Adding device header files
+
+Copy the [CMSIS header files](https://www.arm.com/products/processors/cortex-m/cortex-microcontroller-software-interface-standard.php)  for the device family into the `ext/` folder. You can usually download them as part of a driver pack from the vendor's website.
+Have a look at [the STM32 device headers](https://github.com/roboterclubaachen/xpcc/tree/develop/ext/st) as an example for what you are looking for.
+
+Create a `README.md` that contains the source and copyright (!) of the header files.
+Don't forget to add the include path of the device files to the [`ext/SConstruct`](https://github.com/roboterclubaachen/xpcc/blob/develop/ext/SConscript) file!
+
+
+### Adding device files
+
+The [xpcc device files](https://github.com/roboterclubaachen/xpcc/tree/develop/src/xpcc/architecture/platform/devices) contain the information that is used to generate the HAL. For AVR and STM32 devices these files are generated from another (vendor specific) data source.
+However, for a new vendor such machine readable data is likely not available, and we recommend building a minimal proof-of-concept port for one device first.
+
+You need to manually create a new device file for your device. We recommend copying and modifying [this device file for the STM32F303](https://github.com/roboterclubaachen/xpcc/blob/develop/src/xpcc/architecture/platform/devices/stm32/stm32f303-c_k_r-6_8.xml) for your device.
+
+Vendors usually have a naming convention of their devices, make sure the device attributes and file name reflect this.
+For example, the `SAM4LC4B` identifier can be split like this:
+```xml
+  <device platform="sam" family="4" name="lc" pin_id="b" size_id="4">
+    <!-- ... -->
+  </device>
+```
+
+Modify the F303 device file to reflect the new devices data.
```

Comment:
argh. This sound almost like "Modify the F303 device file _in place_ ..." which is not what you are intended to do here.